src/extension.ts

```diff
@@ -252,10 +285,23 @@ export async function _runAndDisplayDfa(statusBarItem: vscode.StatusBarItem, run
 		vscode.window.showErrorMessage(messages.error.analysisFailedGenerator(errMsg));
 		outputChannel.error(errMsg)
 		outputChannel.show();
-		statusBarItem.hide();
 	}
 }
 
+export async function _shouldProceedWithDfaRun(context: vscode.ExtensionContext, channel: vscode.LogOutputChannel): Promise<boolean> {
+	if (context.workspaceState.get(Constants.GLOBAL_DFA_PROCESS)) {
+		await vscode.window.showInformationMessage(messages.graphEngine.stopDfaRunConfirmationText, messages.graphEngine.stopDfaRunConfirmationYes, messages.graphEngine.stopDfaRunConfirmationNo)
+			.then(async answer => {
+				if (answer === messages.graphEngine.stopDfaRunConfirmationYes) {
+					await _stopExistingDfaRun(context, channel);
+					return true;
```

Comment:
Why does this callback return `true` here? The return value isn't used anywhere as far as I can tell. Also, right now there are cases where it won't return anything, which seems like it's a problem.

Reply:
Agreed, this should be here.